dbfit-java/core/src/main/java/dbfit/util/Binary.java

```diff
@@ -0,0 +1,70 @@
+package dbfit.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Binary {
+    private byte[] binaryData;
+
+    public Binary(final byte[] byteStream) {
+        binaryData = byteStream;
+    }
+
+    public int getLength() {
+        return binaryData.length;
+    }
+
+    private byte getByte(int index) {
+        return binaryData[index];
+    }
+
+    public byte[] getBytes() {
+        return binaryData;
+    }
+
+    public static Object parse(String s) throws Exception {
+        List<Byte> bytes = new ArrayList<Byte>();
+        if (!s.substring(0, 2).equals("0x")) {
+            throw new NumberFormatException("Hexadecimal string does not start with 0x");
+        }
+        String input = s.substring(2);
+        for (int i = ((input.length()) / 2); i > 0; i--) {
+            int index = (i * 2) - 2;
+            int j = Integer.parseInt(input.substring(index, index + 2), 16);
+            bytes.add((byte)j);
+        }
+        byte[] binaryData = new byte[bytes.size()];
+        for (int i = 0; i < bytes.size(); i++) {
+            binaryData[i] = bytes.get(i);
+        }
+        return new Binary(binaryData);
+    }
+
+    @Override
+    public String toString() {
+        String out = "0x";
+        for (int i = binaryData.length - 1; i >= 0; i--) {
+            out = out + String.format("%02X", binaryData[i]);
+        }
+        return out;
+    }
+
+    @Override
+    public boolean equals(final Object other) {
```

Comment:
That's similar to `Arrays.equals(byte[], byte[])` I think so I got wondering - maybe we can delegate to it?

Reply:
Yes. Good shout. Update applied.